speech/grpc/src/main/java/com/google/cloud/speech/grpc/demos/AsyncRecognizeClient.java

```diff
@@ -0,0 +1,248 @@
+  public void recognize() {
+    RecognitionAudio audio;
+    try {
+      audio = createRecognitionAudio();
+    } catch (IOException e) {
+      logger.log(Level.WARNING, "Failed to read audio uri input: " + input);
+      return;
+    }
+    logger.info("Sending " + audio.getContent().size() + " bytes from audio uri input: " + input);
+    RecognitionConfig config = RecognitionConfig.newBuilder()
+        .setEncoding(AudioEncoding.LINEAR16)
+        .setSampleRate(samplingRate)
+        .build();
+    AsyncRecognizeRequest request = AsyncRecognizeRequest.newBuilder()
+        .setConfig(config)
+        .setAudio(audio)
+        .build();
+
+    Operation operation;
+    Operation status;
+    try {
+      operation = stub.asyncRecognize(request);
+
+      //Print the long running operation handle
+      logger.log(Level.INFO, String.format("Operation handle: %s, URI: %s", operation.getName(),
+            input.toString()));
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+
+    while (true) {
+      try {
+        logger.log(Level.INFO, "Waiting 2s for operation, {0} processing...", operation.getName());
+        Thread.sleep(2000);
+        GetOperationRequest operationReq = GetOperationRequest.newBuilder()
+            .setName(operation.getName())
+            .build();
+        status = statusStub.getOperation(
+            GetOperationRequest.newBuilder()
+                .setName(operation.getName())
+                .build()
+                );
+
+        if (status.getDone()) {
+          break;
+        }
+      } catch (Exception ex) {
+        logger.log(Level.WARNING, ex.getMessage());
+      }
+    }
+
+    try {
+      AsyncRecognizeResponse asyncRes = status.getResponse().unpack(AsyncRecognizeResponse.class);
+
+      logger.info("Received response: " + asyncRes);
+    } catch (com.google.protobuf.InvalidProtocolBufferException ex) {
+      logger.log(Level.WARNING, "Unpack error, {0}",ex.getMessage());
+    }
+  }
+
+  public static void main(String[] args) throws Exception {
+
+    String audioFile = "";
+    String host = "speech.googleapis.com";
+    Integer port = 443;
+    Integer sampling = 16000;
+
+    CommandLineParser parser = new DefaultParser();
+
+    Options options = new Options();
+    options.addOption(OptionBuilder.withLongOpt("uri")
+        .withDescription("path to audio uri")
+        .hasArg()
+        .withArgName("FILE_PATH")
+        .create());
+    options.addOption(OptionBuilder.withLongOpt("host")
+        .withDescription("endpoint for api, e.g. speech.googleapis.com")
+        .hasArg()
+        .withArgName("ENDPOINT")
+        .create());
+    options.addOption(OptionBuilder.withLongOpt("port")
+        .withDescription("SSL port, usually 443")
+        .hasArg()
+        .withArgName("PORT")
+        .create());
+    options.addOption(OptionBuilder.withLongOpt("sampling")
```

Comment:
It seems like FILE_PATH and RATE are the only two of these most users will be interested in, why are the others here?